core/drivers/zynqmp_csu_puf.c

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2021 Foundries.io Ltd.
+ *
+ * Jorge Ramirez-Ortiz <jorge@foundries.io>
+ */
+#include <drivers/zynqmp_csu.h>
+#include <drivers/zynqmp_csu_aes.h>
+#include <drivers/zynqmp_csu_puf.h>
+#include <initcall.h>
+#include <io.h>
+#include <kernel/delay.h>
+#include <mm/core_memprot.h>
+
+#define PUF_CMD_OFFSET			0x00
+#define PUF_CFG0_OFFSET			0x04
+#define PUF_CFG1_OFFSET			0x08
+#define PUF_SHUT_OFFSET			0x0C
+#define PUF_STATUS_OFFSET		0x10
+#define PUF_WORD_OFFSET			0x18
+
+#define PUF_REGISTRATION		1
```

Comment:
This goes with term `key registration`.

Reply:
sorry what do you mean? Key registration is different from PUF registration. This is about registering the PUF. But I am removing it anyway - PUF registration must be done using the Xilinx tools still.